examples/soap-calculator/test/acceptance/application.acceptance.ts

```diff
@@ -0,0 +1,80 @@
+import {createClientForHandler, supertest} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {SoapCalculatorApplication} from '../../src/application';
+import {expect} from '@loopback/testlab';
+import {givenAConnectedDataSource} from '../helpers';
+
+describe('Application', function() {
+  let app: SoapCalculatorApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+
+  // tslint:disable-next-line:no-invalid-this
+  this.timeout(5000);
+
+  before(givenAnApplication);
+  before(givenARestServer);
+  before(async () => {
+    await givenAConnectedDataSource();
+  });
+
+  before(async () => {
+    await app.boot();
+    await app.start();
+  });
+
+  before(() => {
+    client = createClientForHandler(server.requestHandler);
+  });
+
+  after(async () => {
+    await app.stop();
+  });
+
+  it('should fail with 412 error', async () => {
+    await client.get('/divide/20/0').expect(412);
+  });
+
+  it('should fail with 404 error', async () => {
+    await client.get('/divide/').expect(404);
+    await client.get('/multiply/').expect(404);
+    await client.get('/add/').expect(404);
+    await client.get('/subtract/').expect(404);
+  });
+
+  it('should divide two numbers', async () => {
+    const response = await client.get('/divide/50/2').expect(200);
+    const answer = {result: {DivideResult: 25}};
+    expect(response.body).to.containDeep(answer);
+  });
+
+  it('should add two numbers', async () => {
+    const response = await client.get('/add/25/25').expect(200);
+    const answer = {result: {AddResult: 50}};
+    expect(response.body).to.containDeep(answer);
+  });
+
+  it('should multipliy two numbers', async () => {
+    const response = await client.get('/multiply/25/3').expect(200);
+    const answer = {result: {MultiplyResult: 75}};
+    expect(response.body).to.containDeep(answer);
+  });
+
+  it('should subtract two numbers', async () => {
+    const response = await client.get('/subtract/100/25').expect(200);
+    const answer = {result: {SubtractResult: 75}};
+    expect(response.body).to.containDeep(answer);
+  });
+
+  function givenAnApplication() {
+    app = new SoapCalculatorApplication({
+      rest: {
+        port: 0,
+      },
+    });
+  }
+
+  async function givenARestServer() {
+    server = await app.getServer(RestServer);
```

Comment:
This should not be needed, you can access the REST server instance via `app.restServer`.